classes.js

```diff
@@ -0,0 +1,310 @@
+
+let outsideTempt = new Celsius(10.0)
+outsideTempt.celsius //returns 10.0
+outsideTempt.getKelvinTemp() //returns 283.0
+outsideTempt.getFahrenheitTemp() //returns 50.0
+
+c.Give Celsius a method called isBelowFreezing that returns a Bool(true
+    if the temperature is below freezing). */
+
+let freezingPoint = {
+    celsius: 0,
+    fahrenheit: 32,
+    kelvin: 273.1
+}
+
+class Celsius {
+    constructor(celsius) {
+        this.celsius = celsius;
+    }
+    getFahrenheitTemp() {
+        return 1.8 * this.celsius + 32
+
+    }
+    getKelvinTemp() {
+        return this.celsius + 273
+    }
+    isBelowFreezing() {
+        if (this.celsius < 0) {
+            return true
+        } else {
+            return false
+        }
+    }
+}
+let outsideTempt = new Celsius(10)
+console.log(outsideTempt.getKelvinTemp()) //returns 283.0
+console.log(outsideTempt.getFahrenheitTemp()) //returns 50.0
+
+/*Question 5
+a.Create a class called Movie that has properties
+for name, year, genre, cast, and description.Create an instance of your Movie
+
+b.Create an method inside Movie called blurb that returns a formatted string describing the movie.
+
+Ex: "Borat came out in 2006. It was an odd film starring Sacha Baron Cohen as a man named Borat who was visiting America from Kazakhstan."*/
+class Movie {
+    constructor(name, year, genre, cast, description) {
+        this.name = name;
+        this.year = year;
+        this.genre = genre;
+        this.cast = cast;
+        this.description = description;
+    }
+    blurb() {
+        return `${this.name} is a movie release ${this.year} about a struggling salesman played by ${this.cast}. ${this.description}.`
+    }
+}
+let movieOne = new Movie("The Pursuit of Happyness", 2006, "drama", "Will Smith", "The Pursuit of Happyness is a movie release in 2006 about a struggling salesman played by Will Smith. He takes custody of his son as he's poised to begin a life-changing professional career.")
+console.log(movieOne.blurb());
+
+/*Question 6
+Write a class Vector that represents a vector in two - dimensional space.It takes two number arguments: x and y parameters, which it should be saved to properties of the same name.
+
+Give the Vector prototype two methods, plus and minus, that take another vector as an argument and returns a new vector that has the sum or difference of the two vectors’(the one in this and the parameter) x and y values.
+
+Add a method getLength to the prototype that computes the length of the vector;
+that is, the distance of the point(x, y) from the origin(0, 0).(a ^ 2 + b ^ 2 = c ^ 2)
+
+Vectors at mathisfun.com
+
+var v1 = new Vector(1, 2)
+var v2 = new Vector(2, 3)
+console.log(v1.plus(v2));
+// => Vector {x: 3, y: 5}
+console.log(v1.minus(v2));
+// => Vector {x: -1, y: -1}
+
+var v3 = new Vector(3, 4)
+console.log(v3.getLength());
+// => 5*/
+function Vector(x, y) {
```

Comment:
These should be written is ES6 class. Not finished inside.

Reply:
Hi Corey, I did not understand this question. Can we please go over it?